service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOService.java

```diff
@@ -0,0 +1,119 @@
+        Map<String, Object> data = new HashMap<>();
+        switch (cmos.size()) {
+            case 0:
+                data.put("numDraftCMOs", "NONE");
+                break;
+            case 1:
+                CaseManagementOrder cmo = cmos.get(0).getValue();
+                data.put("numDraftCMOs", "SINGLE");
+                data.put("reviewCMODecision",
+                    ReviewDecision.builder().hearing(cmo.getHearing()).document(cmo.getOrder()).build());
+                break;
+            default:
+                data.put("numDraftCMOs", "MULTI");
+                data.put("cmoToReviewList", buildDynamicList(cmos));
+                break;
+        }
+
+        return data;
+    }
+
+    public Element<CaseManagementOrder> getCMOToSeal(CaseData caseData, Element<CaseManagementOrder> cmo) {
+        Element<HearingBooking> cmoHearing = getCmoHearingFromId(caseData.getHearingDetails(), cmo.getId());
+        DocumentReference order;
+
+        if (JUDGE_AMENDS_DRAFT.equals(caseData.getReviewCMODecision().getDecision())) {
+            order = caseData.getReviewCMODecision().getJudgeAmendedDocument();
+        } else {
+            order = cmo.getValue().getOrder();
+        }
+        return element(sealFrom(order, cmoHearing.getValue(), time.now().toLocalDate()));
+    }
+
+    private Element<HearingBooking> getCmoHearingFromId(List<Element<HearingBooking>> hearings, UUID cmoId) {
+        return hearings
+            .stream()
+            .filter(hearing -> cmoId.equals(hearing.getValue().getCaseManagementOrderId()))
+            .findFirst()
+            .orElseThrow(NoHearingBookingException::new);
+    }
+
+    public List<Element<CaseManagementOrder>> getCMOsReadyForApproval(List<Element<CaseManagementOrder>> draftCMOs) {
+        return draftCMOs.stream().filter(cmo -> cmo.getValue().getStatus().equals(SEND_TO_JUDGE)).collect(
+            Collectors.toList());
+    }
+
+    public Element<CaseManagementOrder> getSelectedCMO(CaseData caseData) {
+        if (("MULTI").equals(caseData.getNumDraftCMOs())) {
+            UUID selectedCMOCode = getSelectedCMOId(caseData.getCmoToReviewList());
+
+            return caseData.getDraftUploadedCMOs().stream()
+                .filter(element -> element.getId().equals(selectedCMOCode))
+                .findFirst()
+                .orElseThrow(() -> new CMONotFoundException("Could not find draft cmo with id " + selectedCMOCode));
+        } else {
+            return caseData.getDraftUploadedCMOs().get(caseData.getDraftUploadedCMOs().size() - 1);
+        }
+    }
+
+    private UUID getSelectedCMOId(Object dynamicList) {
+        //see RDM-5696 and RDM-6651
+        if (dynamicList instanceof String) {
+            return UUID.fromString(dynamicList.toString());
+        }
+        return mapper.convertValue(dynamicList, DynamicList.class).getValueCode();
+    }
+
+    public CaseManagementOrder getLatestSealedCMO(List<Element<CaseManagementOrder>> cmos) {
```

Comment:
tbh this method is not right, there is nothing special that makes it 'sealed' related. You should pass CaseData as param, then inside method caseData.getSealedCmo(), and then this make it 'sealed' related.

it simplify clients as well, client says , here is a case, give me lates sealed cmo
currently client says: here is list of cmos, give me lates one (thus method should be named getLatestCmo)

To be honest, most of public method should accept CaseData and extract relevant information. It makes class encapsulated better (if you change filed name in CaseData you dont need to amend interface of service not controller). Additionally it makes client simpler and dump, controlled does no need to know how to get e.g sealed cms, it as service, and service knows